dist/responsive_images.js

```diff
@@ -499,6 +510,90 @@ function getImgData( img ) {
     Object.values( imgData.intrinsicOrExtrinsicSizing ).includes( 'intrinsic' )
   );
 
+  // what was the impact of incorrect sizes values?
+
+  // helpers...
+  function geometricDistance(a, b) {
+    if (a > b) {
+      return a / b;
+    } else {
+      return b / a;
+    }
+  }
+
+  const smallestResourceThatsLarger = function( annotatedCandidates ) {
+    const distances = annotatedCandidates.map( c => c.linearDistance );
+    const larger = distances.filter( (d) => { return d >= 0 } );
+    let selectedDistance;
+    if ( larger.length === 0 ) {
+      selectedDistance = Math.max( ...distances );
+    } else {
+      selectedDistance = Math.min( ...larger );
+    }
+    return annotatedCandidates.find( c => c.linearDistance === selectedDistance );
+  }
+
+  const closestResource = function( annotatedCandidates ) {
+    const distances = annotatedCandidates.map( c => c.geometricDistance );
```

Comment:
`return annotatedCandidates.sort((a, b) => {return a.geometricDistance - b.geometricDistance;})[0]` assuming annotatedCandidates is not empty (we should probably validate that assumption)

Reply:
Empty annotatedCandidates is proobbbably not possible because of https://github.com/eeeps/custom-metrics/blob/016611994161bc19cca6d2b5020ae3b19eb13d3e/dist/responsive_images.js#L553, but I'll add checks. And yeah your logic is cleaner / more standard I think...